app/src/main/java/com/mithridat/nonoconverter/backend/solver/NonogramSolver.java

```diff
@@ -218,4 +243,65 @@ boolean applyDualPosition(int ind, int type) {
         return false;
     }
 
+    /**
+     * For every row or column with index in set applies algorithms
+     *
+     * @param set - set of indexes
+     * @param type - ROW, if row
+     *               COL, if column
+     * @return true, if any cell was filled
+     *         false, otherwise
+     */
+    private boolean applyAlgorithms(HashSet<Integer> set, int type) {
+        boolean isFilled = false, isChanged;
+        for (int i : set) {
+            if (_asyncTask.isCancelled()) {
+                break;
+            }
+            isChanged = true;
+            while (isChanged) {
+                isChanged = applySimpleBoxes(i, type);
+                isChanged = applySimpleSpaces(i, type) || isChanged;
+                isChanged = applyForcing(i, type) || isChanged;
+                isChanged = applyGlue(i, type) || isChanged;
+                isChanged = applyJoining(i, type) || isChanged;
+                isChanged = applySplitting(i, type) || isChanged;
+                isChanged = applyPunctuating(i, type) || isChanged;
+                isChanged = applyDualPosition(i, type) || isChanged;
+                isFilled = isChanged || isFilled;
+            }
+        }
+        return isFilled;
+    }
+
+    /**
+     * Method for initialising one of the two boolean arrays
+     *
+     * @param array - array for initialising
+     * @param type - ROW, if _left
+     *               COL, if _top
+     */
+    private void initBooleanArray(boolean[][] array, int type) {
+        int length;
+        for (int i = 0; i < array.length; i++) {
+            length = type == Field.ROW
+                    ? _nono.getLeftRowLength(i) : _nono.getTopColLength(i);
+            array[i] = new boolean[length];
+            for (int j = 0; j < length; j++) {
+                array[i][j] = false;
+            }
+        }
+    }
+
+    /**
+     * Method for initialising one of the two hash sets
+     *
+     * @param set - hash set for initialising
+     * @param size - size of hash set
+     */
+    private void initHashSet(HashSet<Integer> set, int size) {
```

Comment:
Is it possible to change this method to method for adding in set numbers from the interval?